transport/src/main/java/io/netty/channel/pool/ChannelPoolMap.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.pool;
+
+
+/**
+ * Allows to map {@link ChannelPool} implementations to a specific key.
+ *
+ * @param <K>   the type of the key
+ * @param <P>   the type of the {@link ChannelPool}
+ */
+public interface ChannelPoolMap<K, P extends ChannelPool> {
+    /**
+     * Return the {@link ChannelPool} for the {@code code}. This will never return {@code null}.
```

Comment:
Can we expand on this comment to indicate that a new pool will be created if one that is not found associated with `key`?